### DISCUSSION TITLE
Renaming and Changing Data Type From Input to Select

### DISCUSSION THREAD

**Discussion by falconsmilie** [pimcore/pimcore#13091] [Q&A / Support] [answered]
Hi. I am new to Pimcore. Please forgive my ignorance.

I want to change an existing data type from Input to Select, and rename the field.

I am able to do this via the admin panel, and can update the template views to use the new field name.

My concern is how to manage this change when pushing to prod. Is there a way to programatically add the options to the select field, while still being editable in the admin panel? It seems prone to failure if I upload new templates which reference field data (select options) which don't yet exist.

**Comment by BlackbitDevs** [comment, marked as answer]
Please have a look at the command `bin/console pimcore:deployment:classes-rebuild`. This can be used to migrate your database during deployment. This way you can change the field type locally (or on a development system). Doing so will update the definition file of your data object class in `<Pimcore root folder>/var/classes/definition_<class id>.php`. Commit this definition file to git. 
Now change your template and other places where the field is used. Also commit this change to Git.

Then deploy your changes. After deployment run `bin/console pimcore:deployment:classes-rebuild`. This will update your database and PHP class.

PS: Be careful when changing the field type. While converting from `input` to `select` is no problem because they are stored equally in the database, converting from `input` to `multiselect` or even complexer conversions like from `select` to a `relation` or `numeric` to `quantity value` will cause complete loss of data in this field. Please have a look at https://www.youtube.com/watch?v=OJKxWtgwi3c&list=PL4-QRNfdsdKIfzQIP-c9hRruXf0r48fjt&index=3&t=3612s on how to keep and migrate data in such cases.
